rfd/0046-database-access-config.md

```diff
@@ -64,14 +66,270 @@ $ clusterawsadm bootstrap iam print-policy --document AWSIAMManagedPolicyControl
+  #   uri: aurora-cluster-1.abcdefghijklmnop.us-west-1.rds.amazonaws.com:5432
+  #   # AWS specific configuration.
+  #   aws:
+  #     # Region the database is deployed in.
+  #     region: us-west-1
+  #     # RDS/Aurora specific configuration.
+  #     rds:
+  #       # Aurora Cluster ID. Only present on Aurora databases.
+  #       cluster_id: aurora-cluster-1
+  # # Redshift database static configuration.
+  # # For more information: https://goteleport.com/docs/database-access/guides/postgres-redshift/
+  # - name: redshift
+  #   description: AWS Redshift cluster configuration example.
+  #   # Supported protocols for Redshift: "postgres" or "mysql"
+  #   protocol: postgres
+  #   # Database connection endpoint. Must be reachable from Database service.
+  #   uri: redshift-cluster-example-1.abcdefghijklmnop.us-west-1.redshift.amazonaws.com:5439
+  #   # AWS specific configuration.
+  #   aws:
+  #     # Region the database is deployed in.
+  #     region: us-west-1
+  #     # Redshift specific configuration.
+  #     redshift:
+  #       # Redshift Cluster ID.
+  #       cluster_id: redshift-cluster-example-1
+  # # Self-hosted static configuration.
+  # - name: self-hosted
+  #   description: Self-hosted database configuration.
+  #   # Supported protocols for self-hosted: postgres, mysql, mongodb, cockroachdb.
+  #   protocol: postgres
+  #   # Database connection endpoint. Must be reachable from Database service.
+  #   uri: database.example.com:5432
+auth_service:
+  enabled: "no"
+ssh_service:
+  enabled: "no"
+proxy_service:
+  enabled: "no"
+```
+
+Similarly to configure command, it will also support writing the configuration
+directly to a file.
+
+```bash
+# write to the default file location.
+$ teleport db configure create --output file
+
+Wrote config to file "/etc/teleport.yaml". Now you can start the server. Happy Teleporting!
+```
+
+```bash
+# write to the provided file location.
+$ teleport db configure create --output file:///teleport.yaml
+
+Wrote config to file "/teleport.yaml". Now you can start the server. Happy Teleporting!
+```
+
+Configure a single database.
+
+```bash
+# generates a configuration for a Postgres database.
+$ teleport db configure create \
+   --token=/tmp/token \
+   --auth-server=localhost:3025 \
+   --name=sample-db \
+   --protocol=postgres \
+   --uri=postgres://localhost:5432
+```
+
+### Bootstrap permissions command
 
-The "configure" command will provide subcommands for configuring database agents
-in different environments. This RFD focuses on `aws` subcommands that deal with
-configuring IAM for agents that proxy AWS-hosted databases (RDS, Aurora,
-Redshift).
+Subcommand will bootstrap the necessary configuration for the database agent. It
+reads the provided agent configuration to determine what will be bootstrapped.
+
+```bash
+$ teleport db configure bootstrap [-c=,--config=] [--mode=] [--policy-name=] [--attach-to=] [--confirm]
```

Comment:
I think that "bootstrap" may make an impression that this command is only suitable for initial agent configuration while we want to make sure people can use it just to update existing agent configuration (e.g. change config + reapply it). Maybe "apply" would be a better sub-command, or do you think it doesn't carry that meaning?

Reply:
I kept it as `bootstrap` because since we have the `create` command, `apply` seems to be a command to update the configuration file (like in `kubectl`) instead of setting up external dependencies. I agree that it might sound like a command that should be executed only once. I considered renaming `create` to `init` (similar to the `terraform` commands) to avoid this confusion. But I'm OK with renaming it to `apply`.